src/execution/utils.ts

```diff
@@ -302,6 +306,14 @@ async function prepareTransactionAsIntent(
     destinationExecutions: calls,
     destinationGasUnits: gasLimit,
     accountAccessList,
+    options: {
+      topupCompact: false,
+      sponsorSettings: {
+        gasSponsored: isSponsored,
+        bridgeFeesSponsored: isSponsored,
+        swapFeesSponsored: isSponsored,
```

Comment:
Not sure if sponsorships should be enabled for swaps by default. At some point we will probably expose each of those separately, but need to think of a good enough default.

Reply:
yeah think this is good for now